src/linux.rs

```diff
@@ -0,0 +1,912 @@
+use crate::{ CtlValue, SysctlError, };
+
+use libc::{ c_int, c_void, c_ulong, size_t, };
+
+use std::fs::OpenOptions;
+use std::io::{ self, Write, Read, };
+
+
+/// Takes the name of the OID as argument and returns
+/// a result containing the sysctl value if success,
+/// or a SysctlError on failure
+///
+/// # Example
+/// ```
+/// extern crate sysctl;
+///
+/// fn main() {
+///     println!("Value: {:?}", sysctl::value("kern.osrevision"));
+/// }
+/// ```
+pub fn value(name: &str) -> Result<CtlValue, SysctlError> {
```

Comment:
The API is
let ctl = Ctl::new("xxx.yyyy")
let value = ctl.value();

You need to follow this on Linux as well so code is portable (except from maybe different string input value)

Reply:
done.